editor.planx.uk/src/@planx/components/FindProperty/Public/index.tsx

```diff
@@ -119,7 +136,11 @@ function Component(props: Props) {
         setRegions([...new Set(regions)]);
       }
     }
-  }, [data]);
+
+    if (geojson?.type) {
+      setBoundary(geojson);
+    }
```

Comment:
I'm not as up to speed on React as I'd like to be (or should be!) but I think this would be best as it's own `useEffect()`. If not, it will re-set the boundary each time `data` changes - we just care about changes to `geojson`

Reply:
you're right! now it's own `useEffect()`. SWR was doing smart caching here though I think so we didn't actually see repeated requests in the network tab as far as I could tell